docs/development/design/new-sphinx-guides.rst

```diff
@@ -0,0 +1,186 @@
+      -  Check that our code works (yay!)
+
+   2. Using virtual environments for development
+
+      -  Create a virtual environment (and/or conda environment)
+      -  Activate our virtual environment (it will always be the first
+         step)
+      -  Check that our code also works (yay!)
+
+   3. Adding external dependencies
+
+      -  Add import from small third party library
+      -  Install it inside our virtual environment
+      -  Check that our code works (yay!) and it doesn’t outside the
+         virtual environment (as expected)
+
+   4. Appendix: Using version control
+
+      -  Install git (we will use it during the tutorial)
+      -  Write a barebones ``README.md`` (we will expand it later), a
+         ``.gitignore`` file (copied from gitignore.io), and a proper
+         ``LICENSE`` file
+
+         -  ``LICENSE.md``? ``LICENSE``? ``COPYING``? Whatever
+
+      -  Create the first commit for the project (yay!)
+
+2. First steps to document our project using Sphinx
+
+   1. Installing Sphinx
+
+      -  Activate our virtual environment
+      -  Install Sphinx inside the virtual environment
+      -  Check that ``sphinx-build --help`` works (yay!)
+
+   2. Creating the project layout
+
+      -  Run the sphinx-quickstart wizard: separate source and build
+         (``y``), project name, author name, and go
+      -  Check that the correct files are created (yay!)
+
+   3. Converting our documentation to local HTML
+
+      -  Build the HTML output using
+         ``sphinx-build -b -W html doc doc/_build/html``  [1]_
+      -  Navigate to ``doc/_build/html`` and launch an HTTP server
+         (``python -m http.server``)
+      -  Open http://localhost:8000 in a web browser, and see the HTML
+         documentation (yay!)
+
+   4. Making a change and regenerating the documentation
+
+      -  Add a sentence to ``index.rst`` (no need to explain in detail
+         what reST is at this stage)
+      -  Rebuild HTML ``sphinx-build -W -b html ...``
+      -  Observe that the HTML docs have changed (yay!) (we will have to
+         do this every time we change something)
+
+   5. Simplify documentation building by using Make
+
+      -  Install Make (and its Windows counterpart)
+      -  Make another trivial change
+      -  Build HTML doing ``cd doc && make html``
+      -  Observe that the HTML docs have changed (yay!)
+
+   6. Converting our documentation to PDF
+
+      -  Install TeX Live to create PDF output (should work on every
+         operative system)
+      -  Build LaTeX using ``make latexpdf``
+      -  See that the PDF appeared (yay!)
+
+   7. Appendix: PDF without LaTeX using rinoh (beta)
+
+3. Customizing Sphinx configuration
+
+   1. Changing the HTML theme
+
+      -  Install https://pypi.org/project/furo/
+      -  Change the ``html_theme`` in ``conf.py``
+      -  Rebuild the HTML documentation and observe that the theme has
+         changed (yay!)
+
+   2. Changing the PDF appearance
```

Comment:
If we want another format, we should do something with a ton less dependencies, like a man page or epub. I'm not sure we want to dive so deeply into customizing anything other than HTML though, based on what I've seen users care about.

Reply:
Yes, the intention was to introduce a second format, so it's clear that it can be done and also where does the configuration go.

In line with my previous comment, I still think PDF is valuable because it's a format everyone is familiar with, and it can be opened either with a web browser or with any PDF viewer. Also, I've seen comments of people turning down other documentations systems because they lacked PDF export, so this is perhaps one of Sphinx strengths, even if HTML is used more. According to our Google Analytics, there is a small but nonzero amount of people downloading PDFs from Read the Docs at least.

As a second option, https://github.com/brechtm/rinohtype supports PDF generation without *TeX - I tested it and works nicely. However, the README says that it's still beta.

Reply:
Just in case, I bit the bullet and tried to build an EPUB locally. The multi-file output was confusing at first, but `pdflatex` has the same issue. I viewed the `.epub` result with https://calibre-ebook.com/ and it worked.

man pages are Linux/UNIX-specific I guess, so I would like to use something more generic. I would reluctantly use EPUB instead of PDF as a secondary format, as long as we keep it at least as an optional section.

Reply:
It adds multiple GB of downloads, at least to do it at the RTD level. It also has a lot of issues with non-english languages, terrible error output, and lots of other drawbacks. I'm definitely not convinced it's a value-add to any tutorial.

What is the goal of showing additional output formats? In my mind, it shows the value of writing in RST and then outputting in another format. If that's the goal, lets just do text output or something.